### PR TITLE
[Merged by Bors] - chore: turn `induction'` into `induction`

### PR DESCRIPTION
**Pull request by pitmonticone**
- [x] turn the (soon to be deprecated) `induction'` into `induction`
- [x] golf a few proofs

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Ring/CentroidHom.lean [hunk 1]
```diff
@@ -270,16 +270,18 @@ instance hasNPowNat : Pow (CentroidHom α) ℕ :=
...
-        induction' n with n ih
-        · exact rfl
-        · rw [pow_succ']
+        induction n with
+        | zero => rfl
+        | succ n ih =>
+          rw [pow_succ']
           exact (congr_arg f.toEnd ih).trans (f.map_mul_right' _ _)
-        }⟩
+    }⟩
```
Comment (joneugster) [inline Mathlib/Algebra/Ring/CentroidHom.lean:284]: I think the style guide says to put ` }⟩` at the end of the last line, but imo not that important in this PR.

**Review by joneugster — APPROVED** [review]
Looks good to me!

maintainer merge

File: Mathlib/Algebra/Ring/CentroidHom.lean (hunk 1 above)
Comment (pitmonticone) [reply Mathlib/Algebra/Ring/CentroidHom.lean:284]: Done. Thanks.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**